web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputer/Setup.tsx

```diff
@@ -238,7 +238,7 @@ function AgentSetup() {
                 await ctx.connectMyComputerService.createRole(rootClusterUri);
               certsReloaded = response.certsReloaded;
             } catch (error) {
-              if (isAccessDeniedError(error)) {
+              if (isTshdRpcError(error, 'PERMISSION_DENIED')) {
```

Comment:
Thanks to your work on gRPC clients and errors we can fix a bug here.

If we just catch all PERMISSION_DENIED errors, we're going to catch retryable PERMISSION_DENIED errors as well.

```typescript
              if (
                isTshdRpcError(error, 'PERMISSION_DENIED') &&
                !error.isResolvableWithRelogin
              ) {
                throw new Error(
                  `Cannot set up the role: ${error.message}. Contact your administrator for permissions to manage users and roles.`
                );
              }
```

<img width="1280" alt="cmc-access-denied" src="https://github.com/gravitational/teleport/assets/27113/7b027db6-40d0-46d2-9cb3-a95726b74a4f">

Reply:
Thanks, fixed!